types/zetapush-js/index.d.ts

```diff
@@ -0,0 +1,184 @@
+  }
+
+  interface WeakClientOptions extends Options {
+    deploymentId?: string;
+  }
+
+  interface Services {
+    Macro: AsyncMacroService;
+    Messaging: Service;
+  }
+
+  class Service {
+    DEFAULT_DEPLOYMENT_ID: string;
+    $publish: ServicePublisher;
+    constructor($publish: ServicePublisher);
+  }
+
+  class AsyncMacroService {
+    DEFAULT_DEPLOYMENT_ID: string;
+    $publish: AsyncMacroServicePublisher;
+    constructor($publish: AsyncMacroServicePublisher);
+  }
+
+  class MacroService {
+    DEFAULT_DEPLOYMENT_ID: string;
+    $publish: MacroServicePublisher;
+    constructor($publish: MacroServicePublisher);
+  }
+
+  interface ServiceDeclaration {
+    Type: Service;
+    deploymentId?: string;
+  }
+
+  interface Token {
+    token: string;
+  }
+
+  interface Credentials {
+    login: string;
+    password: string;
+  }
+
+  interface ClientHelper {
+    authentication: Handshake.AuthenticationCallback;
+    servers: Promise<string[]>;
+    getUniqRequestId(): string;
+  }
+
+  type ConnectionStatusHandler = number;
+}
+
+export interface ConnectionStatusListener {
+  onConnectionBroken(): void;
+  onConnectionClosed(): void;
+  onConnectionEstablished(): void;
+  onConnectionToServerFail(failure: any): void;
+  onConnectionWillClose(): void;
+  onFailedHandshake(failure: any): void;
+  onMessageLost(): void;
+  onNoServerUrlAvailable(): void;
+  onSuccessfulHandshake(authentication: any): void;
+}
+
+export class Authentication {
+  static delegating({ token }: Handshake.TokenAuthData): Handshake.TokenHandshake;
+  static simple({ login, password }: Handshake.CredentialsAuthData): Handshake.CredentialsHandshake;
+  static weak({ token }: Handshake.TokenAuthData): Handshake.TokenHandshake;
+}
+
+export class Client {
+  helper: Helper.ClientHelper;
+  constructor(options: Helper.ClientOptions);
+  addConnectionStatusListener(listener: ConnectionStatusListener): Helper.ConnectionStatusHandler;
+  connect(): void;
+  createService(declaration: Helper.ServiceDeclaration): Helper.Service;
+  createAsyncMacroService(declaration: Helper.ServiceDeclaration): Helper.AsyncMacroService;
+  disconnect(): void;
+  isConnected(): boolean;
+  getSandboxId(): string;
+  getResource(): string;
+  getUserId(): string;
+  removeConnectionStatusListener(listener: Helper.ConnectionStatusHandler): void;
+  setAuthentication(authentication: Handshake.AuthenticationCallback): void;
+  setLogLevel(level: string): void;
+  setResource(resource: string): void;
+  unsubscribe(service: Helper.Service): void;
+}
+
+export class WeakClient extends Client {
+  constructor(options: Helper.WeakClientOptions);
+  getToken(): Helper.Token;
+}
+
+export class SmartClient extends Client {
+  getCredentials(): any;
+  getSession(): any;
+  hasCredentials(): boolean;
+  isStronglyAuthenticated(session?: any): boolean;
+  isWeaklyAuthenticated(session?: any): boolean;
+  setCredentials(credentials: any): void;
+}
+
+export const services: Helper.Services;
```

Comment:
After reviewing the documentation, I would also recommend you add an `export as namespace ZetaPush;` to the file, for those using it via CDN (https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/Modules.md#umd-modules).